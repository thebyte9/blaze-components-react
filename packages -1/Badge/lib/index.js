"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var react_1 = __importDefault(require("react"));
var Badge = function (_a) {
    var children = _a.children, type = _a.type, pill = _a.pill, icon = _a.icon, round = _a.round, to = _a.to, link = _a.link, attrs = __rest(_a, ["children", "type", "pill", "icon", "round", "to", "link"]);
    var assignType = type ? "badge--" + type : "";
    var isPill = pill ? "badge--pill" : "";
    var isRound = round ? "badge--round" : "";
    var withIcon = icon ? "badge--icon-text" : "";
    var classes = "badge " + assignType + " " + isRound + " " + isPill + " " + withIcon;
    return link ? (react_1["default"].createElement("a", __assign({ href: to, className: classes }, attrs), children)) : (react_1["default"].createElement("span", __assign({ className: classes }, attrs), children));
};
Badge.defaultProps = {
    children: "No content",
    icon: false,
    link: false,
    pill: false,
    round: false,
    to: "#",
    type: ""
};
exports["default"] = Badge;
