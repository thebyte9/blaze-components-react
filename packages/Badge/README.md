## Description

Badges are small components typically used to communicate a numerical value or indicate the status of an item to the user.

## Usage

- Simple

```js
<Badge type="primary">Badge</Badge>
```

- Link

```js
<Badge type="info" link>
    <a href="https://www.thebyte9.com/">Go to byte9</a>
</Badge>
```

- Icon

```js
<Badge type="pagebuilder" icon colour="deep-orange">
    Social follow            
  <i className="fas fa-share" />
</Badge>
```

## API

##### Badges can receive a number of `props` as follow:

|  NAME      |            TYPE            |   DEFAULT    |  Options                                                                                   |
| :--------- | :------------------------: | :----------: | ------------------------------------------------------------------------------------------ |
|  type      |           String           |    empty     |  default, primary, secondary, success, alert, info, light, dark, link, pagebuilder         |
|  link      |          Boolean           |    false     |
|  icon      |          Boolean           |    false     |
|  color     |           String           |    empty     |  blue, light-blue, deep-orange, orange, purple, deep-purple, pink, red, cyan, teal, green  |
|  children  |  Single or array of nodes  |  No content  |
