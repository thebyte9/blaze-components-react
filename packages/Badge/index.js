import React from 'react';
import PropTypes from 'prop-types';

const Badge = ({ children, type, pill, icon, round, to, link, ...attrs }) => {
  const assignType = type ? `badge--${type}` : '';
  const isPill = pill ? 'badge--pill' : '';
  const isRound = round ? 'badge--round' : '';
  const withIcon = icon ? 'badge--icon-text' : '';
  const classes = `badge ${assignType} ${isRound} ${isPill} ${withIcon}`;

  return link ? (
    <a href={to} className={classes} {...attrs}>
      {children}
    </a>
  ) : (
    <span className={classes} {...attrs}>
      {children}
    </span>
  );
};

Badge.propTypes = {
  type: PropTypes.string,
  to: PropTypes.string,
  round: PropTypes.bool,
  link: PropTypes.bool,
  pill: PropTypes.bool,
  icon: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};

Badge.defaultProps = {
  type: '',
  to: '#',
  round: false,
  pill: false,
  link: false,
  icon: false,
  children: 'No content'
};

export default Badge;
