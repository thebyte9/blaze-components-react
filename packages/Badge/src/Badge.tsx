import buildClassNames from '../../Utils/src/buildClassNames';
import React, { Fragment, FunctionComponent } from "react";

interface IBadgeProps {
  type?: string;
  link?: boolean;
  icon?: boolean;
  color?: string;
  modifiers?: string;
  children?: React.ReactNode;
}

const Badge: FunctionComponent<IBadgeProps> = ({
  children,
  type,
  color,
  icon,
  link,
  modifiers,
  ...attrs
}) => {
  const withIcon = icon ? "badge--icon-text" : "";

  const classes: string = buildClassNames("badge", {
    [`badge--${type}`]: !!type,
    [`${withIcon}`]: !!withIcon,
    [`${color}`]: !!color,
    [`${modifiers}`]: !!modifiers,
  });

  return link ? (
    <Fragment>{children}</Fragment>
  ) : (
    <span className={classes} {...attrs}>
      {children}
    </span>
  );
};

Badge.defaultProps = {
  children: "No content",
  color: "",
  icon: false,
  link: false,
  modifiers: "",
};

export default Badge;
