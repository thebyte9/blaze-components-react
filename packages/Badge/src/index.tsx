import withUtils from '@blaze-react/utils';
import React, { Fragment, FunctionComponent } from 'react';

interface IBadgeProps {
  type?: string;
  link?: boolean;
  icon?: boolean;
  color?: string;
  modifiers?: string;
  children?: any;
  utils: {
    classNames: (className: string | object, classNames?: object) => string;
  };
}
const Badge: FunctionComponent<IBadgeProps> = ({
  children,
  type,
  color,
  icon,
  link,
  utils: { classNames },
  modifiers,
  ...attrs
}) => {
  const withIcon = icon ? 'badge--icon-text' : '';
  const classes: string = classNames('badge', {
    [`badge--${type}`]: !!type,
    [`${withIcon}`]: !!withIcon,
    [`${color}`]: !!color,
    [`${modifiers}`]: !!modifiers,
  });

  return link ? (
    <Fragment>{children}</Fragment>
  ) : (
    <span className={classes} {...attrs}>
      {children}
    </span>
  );
};
Badge.defaultProps = {
  children: 'No content',
  color: '',
  icon: false,
  link: false,
  modifiers: '',
};
export default withUtils(Badge);
