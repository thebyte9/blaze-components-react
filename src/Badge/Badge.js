import React from 'react';
import PropTypes from 'prop-types';

const Badge = ({
  children,
  type,
  pill,
  icon,
  round,
  to,
  link,
  ...attrs
}) => {
  const assignType = type ? `badge--${type}` : '';
  const isPill = pill ? 'badge--pill' : '';
  const isRound = round ? 'badge--round' : '';
  const withIcon = icon ? 'badge--icon-text' : '';
  const classes = `badge ${assignType} ${isRound} ${isPill} ${withIcon}`;

  return (link ? (
    <a href={to} className={classes} {...attrs}>{children}</a>
  ) : (
    <span className={classes} {...attrs}>
      {children}
    </span>
  ));
};

Badge.propTypes = {
  round: PropTypes.bool,
  link: PropTypes.bool,
  pill: PropTypes.bool,
  icon: PropTypes.bool,
  type: PropTypes.string,
  to: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};

Badge.defaultProps = {
  type: '',
  round: false,
  pill: false,
  link: false,
  icon: false,
  to: '#',
  children: 'No content'
};

export default Badge;
