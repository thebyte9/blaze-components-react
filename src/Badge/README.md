
## Description

Badges are small components typically used to communicate a numerical value or indicate the status of an item to the user.

## Usage

* Simple

```js
<Badge type="primary">Badge</Badge>
```

* Pill

```js
<Badge type="success" pill>Badge</Badge>
```

* Link

```js
<Badge type="info" link to="https://www.thebyte9.com/">Link to byte9</Badge>
```

* Rounded

```js
<Badge type="alert" round>10</Badge>
```

## API

##### Badges can receive a number of `props` as follow:


| NAME   | TYPE | DEFAULT | Description |
| :---  | :---:  | :---: | ------- |
| round | Boolean | false | Is the badge round? |
| link | Boolean | false | is the badge a link? |
| pill | Boolean | false | Is is styled as a pill? |
| icon | Boolean | false | Does the badge contain an icon? |
| type | String | '' | primary, secondary, success, warning, dark, info, light, cta |
| to | String | '' | If badge is a link, where to link to |
| children | Single or array of nodes | No content | Any children components within the badge |

