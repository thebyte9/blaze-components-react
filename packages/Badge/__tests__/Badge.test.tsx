import { shallow } from "enzyme";
import expect from "expect";
import React from "react";
import Badge from "../src";

describe("Badge component", () => {
  test("should render children when badge type is link", () => {
    const wrapper = shallow(
      <Badge type="pagebuilder" link>
        <a href="#"></a>
      </Badge>
    );
    const link = wrapper.find("a");
    expect(link.length).toBe(1);
  });

  test("should render children within span when badge type is not link", () => {
    const wrapper = shallow(
      <Badge type="primary">
        <a href="#"></a>
      </Badge>
    );
    const span = wrapper.find("span");
    expect(span.length).toBe(1);
  });

  test("should render correct class when badge type is not link", () => {
    const wrapper = shallow(
      <Badge type="pagebuilder" icon color="deep-orange">
        Social follow         
        <i className="fas fa-share" />
      </Badge>
    );
    const span = wrapper.find("span");
    expect(span.prop("className")).toContain("pagebuilder");
    expect(span.prop("className")).toContain("badge--icon-text");
    expect(span.prop("className")).toContain("deep-orange");
  });

  test("should render badge type icon and badge type text", () => {
    const wrapper = shallow(
      <Badge type="pagebuilder" color="blue" icon>
        BadgeText         
        <i className="fas fa image" />
      </Badge>
    );
    const badge = wrapper.find("span");
    expect(badge.text()).toContain("BadgeText");
    const icon = wrapper.find("i");
    expect(icon.length).toBe(1);
  });

  test("should display correct class based on props", () => {
    const wrapper = shallow(
      <Badge icon>
        BadgeText
        <i className="fas fa-follow" />
      </Badge>
    );
    const span = wrapper.find("span");
    expect(span.prop("className")).toContain("badge--icon-text");
  });
});
