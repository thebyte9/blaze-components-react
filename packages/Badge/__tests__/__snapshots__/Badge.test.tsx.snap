// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component should render children without throwing error 1`] = `
<Badge
  color="blue"
  icon={true}
  link={false}
  modifiers=""
  type="pagebuilder"
  utils={
    Object {
      "ErrorMessage": [Function],
      "buildClassNames": [Function],
      "removeExtraSpaces": [Function],
      "uniqueId": [Function],
    }
  }
>
  BadgeText
  <i
    className="fas fa image"
  />
</Badge>
`;

exports[`Badge component should render properly when pass link prop without throwing error 1`] = `
<Badge
  color=""
  icon={false}
  link={true}
  modifiers=""
  type="pagebuilder"
  utils={
    Object {
      "ErrorMessage": [Function],
      "buildClassNames": [Function],
      "removeExtraSpaces": [Function],
      "uniqueId": [Function],
    }
  }
>
  <a
    href="#"
  />
</Badge>
`;
